src/synthesizer/particle/stars.py

```diff
@@ -641,8 +984,57 @@ def _parametric_young_stars(
         else:
             stars = stars[0]
 
-        # Get the spectra for this parametric form
-        return stars.generate_lnu(grid=grid, spectra_name=spectra_name)
+        self._parametric_young_stars = stars
+
+        # Create index pairs for the SFZH
+        index_pairs = np.asarray(
+            [
+                [[j, i] for i in np.arange(len(grid.metallicity))]
+                for j in np.arange(len(grid.log10age))
+            ]
+        )
+
+        # Find the grid indexes on the parametric grid
+        grid_indexes = index_pairs[stars.sfzh > 0]
+
+        # Create new particle stars object from non-empty SFZH entries
+        new_stars = self.__class__(
+            stars.sfzh[stars.sfzh > 0] * Msun,
+            10 ** grid.log10ages[grid_indexes[:, 0]] * yr,
```

Comment:
FYI @christopherlovell, found a bug here. This was previously `grid.log10ages[grid_indexes[:, 0]]` but particle stars take pure ages.